python-lib/nlp/ontology_tagging/ontology_tagger.py

```diff
@@ -141,14 +143,22 @@ def _set_log_level(self, languages: List[AnyStr]) -> None:
             logger.setLevel(logging.ERROR)
 
     def _remove_incomplete_rows(self) -> None:
-        """Remove rows with at least one empty value from ontology df"""
+        """Remove rows with at least one empty value in the keyword column and the tag column from ontology df"""
         self.ontology_df.replace("", float("nan"), inplace=True)
-        self.ontology_df.dropna(inplace=True)
-        if self.ontology_df.empty:
+        self.ontology_df.dropna(
+            subset=[self.keyword_column, self.tag_column], inplace=True
+        )
+        if self.ontology_df[[self.keyword_column, self.tag_column]].empty:
             raise ValueError(
                 "No valid tags were found. Please specify at least a keyword and a tag in the ontology dataset, and re-run the recipe"
             )
 
+    def _replace_missing_categories(self) -> None:
```

Comment:
Are we sure it's the right behavior to enforce?
Could you give an example of output in that case?

Reply:
As I see it, there are two ways to handle missing categories:
- replace all missing categories with the same "default" one, but it is hard to find the right name for it 
- replace all missing categories with their tags, so the tags are the categories, but it creates non necessary categories.
Krishna have voted for the second option, but it is open to discussion 

Here is how it looks like 
<img width="653" alt="Capture d’écran 2021-06-08 à 19 37 35" src="https://user-images.githubusercontent.com/77485161/121232824-439bda80-c892-11eb-8b02-924b2c888de3.png">
<img width="736" alt="Capture d’écran 2021-06-08 à 19 41 26" src="https://user-images.githubusercontent.com/77485161/121232831-44347100-c892-11eb-8bb1-28e2043d75fc.png">

Reply:
I understand it's difficult to find the name for the "default" category.
However, in your screenshot, isn't it a bit weird to have a column named "politics" that will always only say "politics"? And same in the JSON, its says "politics": ["politics"]. If I were to try thinking like a new user, I would think it is a bug. 
Do you see what I mean?
If we were to put all cases without category in a single "uncategorized" category, would it look better?

Reply:
That's a good idea, thank you 👍 fc93052